app/controllers/concerns/users/time_zone.rb

```diff
@@ -12,5 +13,14 @@ def browser_time_zone
     rescue TZInfo::UnknownTimezone, TZInfo::InvalidTimezoneIdentifier
       Time.zone
     end
+
+    def to_user_timezone(time_date)
+      return "" if time_date.nil? || (time_date.instance_of?(String) && time_date.empty?)
+
+      time_zone = (browser_time_zone && browser_time_zone != Time.zone) ? browser_time_zone : "Eastern Time (US & Canada)"
+      return time_date.in_time_zone(time_zone) if time_date.respond_to?(:in_time_zone)
+
+      time_date.to_time(time_zone)
+    end
```

Comment:
I think this might be doing too much. It looks like it is both finding the time zone to convert to AND doing the conversion. It might make more sense to do something like `to_user_timezone(time_date, time_zone: user_timezone)` and have user timezone by its own method. 

Largely I think this should be split up.

Reply:
Thanks for suggesting. You're right, it does two things in the same method.  
I split it into two methods and added the `user_timezone` method.
You suggested adding a default parameter, but then the function would not return only the user's timezone, instead, any timezone we specify (maybe we need to rename it in this case).